File: 5ku/FirstNonRepeatingCharacter/FirstNonRepeatingCharacter.go
package main

import "strings"

func FirstNonRepeating(str string) string {
	//your code here
	low := strings.ToLower(str)
	for k, _ := range low {
		if strings.Count(low, low[k:k+1]) == 1 {
			return str[k : k+1]
		}

	}
	return ""
}

/*	best
package kata
import (
	"strings"
)
func FirstNonRepeating(str string) string {
    for _, c := range str {
        if strings.Count(strings.ToLower(str), strings.ToLower(string(c))) < 2 {
	          return string(c)
	      }
    }
    return ""
}
*/

File: 5ku/FirstNonRepeatingCharacter/FirstNonRepeatingCharacter.py
def first_non_repeating_letter(s):
    for i in range(len(s)):
        if s.lower().count(s.lower()[i])==1:
            return s[i]
    return ""

""" best
def first_non_repeating_letter(string):
    string_lower = string.lower()
    for i, letter in enumerate(string_lower):
        if string_lower.count(letter) == 1:
            return string[i]
            
    return ""
 """
